### DISCUSSION TITLE
Can i show in Blade

### DISCUSSION THREAD

**Discussion by mchisnerman3** [maize-tech/laravel-markable#10] [Q&A] [answered]
Hi, thanks for your awesome trait.
My question is:

How can i show in blade if a model has like?
Example: $model->HasLike();

Please forgive my poor english

**Comment by riccardodallavia** [comment, marked as answer]
Hi @mchisnerman3 !
Sure, you can do that in many ways I think. A solution I came up with is something like this:

```php
$model
  ->likers()
  ->whereKey($user->getKey())
  ->exists();

$model
  ->likers()
  ->whereKey(auth()->id())
  ->exists();
```

Alternatively, you could implement an accessor in your markable model:

```php
public function getHasLikeAttribute()
{
  return $this
    ->likers()
    ->whereKey(auth()->id())
    ->exists();
}
```

and use it in your blade component:

```php
$model->has_like;
```

**Beware**: using the accessor will lead you to the N+1 problem when working with multiple models at once.
In that case, you could easily append the has_like attribute while executing the query:

```php
$models = Model::query()
  ->withCount([
    'likers as has_like' => fn ($query) => $query->whereKey(auth()->id())
  ])
  ->withCasts([
	'has_like' => 'boolean'
  ])
  ->get();
```

Hope this can help you!
